include/IECore/StringAlgo.inl

```diff
@@ -241,6 +241,10 @@ void tokenize( const std::string &s, const char separator, OutputContainer &outp
 	tokenize<typename OutputContainer::value_type>( s, separator, std::back_inserter( outputContainer ) );
 }
 
+#ifdef _MSC_VER
+#pragma warning( disable:4701 )
+#endif
+
```

Comment:
I'm intrigued by this one - what is the type of `std::iterator_traits<Iterator>::value_type` when MSVC complains?

Reply:
It's a `float` from `StringAlgoTest.cpp` https://github.com/ImageEngine/cortex/blob/bc41fb9442dc434f076ded59eee95059d7cc23f8/test/IECore/StringAlgoTest.cpp#L142-L150

I think a better solution is to initialize it properly instead of ignoring the warning. I reverted the change in https://github.com/ImageEngine/cortex/pull/1217/commits/6e98606ca1c1b8d6e581176653a4c39865f8518b and initialized it in https://github.com/ImageEngine/cortex/pull/1217/commits/37ada7b30b09b38c24145f47db72a5a095efcf3a and MSVC is happy again.

Reply:
I have absolutely no idea why we have tests asserting that `StringAlgo::join()` works for floats, or why we're not just using `boost::algorithm::join()` everywhere - do you know @andrewkaufman?

The initialization solution seems good to me, but I think we should remove this function for 10.4.

Reply:
I have no recollection of this and not much to go on other than the 13 year old commit message "for backwards compatibility with boost 1.33.1 (used by Houdini)." I also don't see any uses of `StringAlgo::join` outside of this test suite.